### PR TITLE
AB#89377 Add green lights view and business logic

### PR DESCRIPTION
**Pull request by dawid-pawulski**
More test to be added in the next pr

image

### CODE REVIEW

File: src/HE.Investment.AHP.WWW/Controllers/SiteController.cs [hunk 1]
```diff
@@ -584,17 +604,10 @@ protected override async Task<IStateRouting<SiteWorkflowState>> Routing(SiteWork
         CancellationToken cancellationToken)
     {
         var siteId = this.GetSiteIdFromRoute();
-        var redirect = ContinueWithRedirect(new { siteId });
-        var action = HttpContext.Request.Form["action"];
-        if (action == GenericMessages.SaveAndReturn)
-        {
-            redirect = Index(cancellationToken);
-        }
-
```
Comment (dawid-pawulski) [inline src/HE.Investment.AHP.WWW/Controllers/SiteController.cs:593]: It is not working correctly. 
I will change it in the next task
[Task 90301](https://dev.azure.com.mcas.ms/homesengland/Investments%20Programme/_workitems/edit/90301?McasTsid=26110&McasCtx=4): Add Extension method for return link

File: src/HE.Investment.AHP.Domain/Site/Entities/SiteEntity.cs [hunk 2]
```diff
@@ -106,11 +110,16 @@ public void ProvideNationalDesignGuidePriorities(NationalDesignGuidePriorities n
...
 
-    public void ProvideBuildingForHealthyLifeCommand(BuildingForHealthyLifeType buildingForHealthyLife)
+    public void ProvideBuildingForHealthyLife(BuildingForHealthyLifeType buildingForHealthyLife)
     {
         BuildingForHealthyLife = _modificationTracker.Change(BuildingForHealthyLife, buildingForHealthyLife);
     }
 
+    public void ProvideNumberOfGreenLights(NumberOfGreenLights? numberOfGreenLights)
+    {
+        NumberOfGreenLights = _modificationTracker.Change(NumberOfGreenLights, numberOfGreenLights);
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/Site/Entities/SiteEntity.cs:120]: probably nullablility check needs to be added to IsAnswered method

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/NumberOfGreenLights.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    public NumberOfGreenLights(string value)
+    {
+        if (!int.TryParse(value, NumberStyles.Integer, CultureInfo.InvariantCulture, out var parsedValue))
+        {
+            OperationResult.New()
+                .AddValidationError(nameof(NumberOfGreenLights), ValidationErrorMessage.MustBeNumber(DisplayName))
+                .CheckErrors();
+        }
+
+        if (parsedValue is < MinValue or > MaxValue)
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/Site/ValueObjects/NumberOfGreenLights.cs:25]: consider using NumericValidator and IsNumber and IsBetween methods

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/Site/Entities/SiteEntity.cs (hunk 2 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.Domain/Site/Entities/SiteEntity.cs:120]: True, I will do that in next pr :)

File: src/HE.Investment.AHP.Domain/Site/ValueObjects/NumberOfGreenLights.cs (hunk 3 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.Domain/Site/ValueObjects/NumberOfGreenLights.cs:25]: Thanks, I will do that in next pr :)

**Pull request merged**